### ISSUE TITLE
Intermittent test failure in PersistentTopicTest.testClosingReplicationProducerTwice

### ISSUE THREAD

**Issue by merlimat** [apache/incubator-pulsar#617]
As seen in https://builds.apache.org/job/pulsar-pull-request/org.apache.pulsar$pulsar-broker/17/testReport/junit/org.apache.pulsar.broker.service/PersistentTopicTest/testClosingReplicationProducerTwice/

### Error Message
```
pulsarClientImpl.createProducerAsync(
    "persistent://prop/global/ns/testClosingReplicationProducerTwice",
    org.apache.pulsar.client.api.ProducerConfiguration@79f3359e,
    "pulsar.repl.local"
);
Wanted 1 time:
-> at org.apache.pulsar.broker.service.PersistentTopicTest.testClosingReplicationProducerTwice(PersistentTopicTest.java:965)
But was 2 times. Undesired invocation:
-> at org.apache.pulsar.broker.service.AbstractReplicator.startProducer(AbstractReplicator.java:114)
```

### Stacktrace
```
org.mockito.exceptions.verification.TooManyActualInvocations: 

pulsarClientImpl.createProducerAsync(
    "persistent://prop/global/ns/testClosingReplicationProducerTwice",
    org.apache.pulsar.client.api.ProducerConfiguration@79f3359e,
    "pulsar.repl.local"
);
Wanted 1 time:
-> at org.apache.pulsar.broker.service.PersistentTopicTest.testClosingReplicationProducerTwice(PersistentTopicTest.java:965)
But was 2 times. Undesired invocation:
-> at org.apache.pulsar.broker.service.AbstractReplicator.startProducer(AbstractReplicator.java:114)

	at org.apache.pulsar.broker.service.PersistentTopicTest.testClosingReplicationProducerTwice(PersistentTopicTest.java:965)
```

**Comment by sijie** [comment]
This is fixed by #1360

**Comment by sijie** [comment]
Fixed by @ivankelly
